src/odemis/acq/test/fastem_test.py

```diff
@@ -437,30 +427,37 @@ def on_progress_update(self, future, start, end):
 
 
 class TestFastEMAcquisitionTask(unittest.TestCase):
-    """Test methods of the fastem.AcquistionTask class."""
+    """Test methods of the fastem.AcquisitionTask class."""
 
     @classmethod
     def setUpClass(cls):
+        cls.mocked_testing = False
+
         if TEST_NOHW:
-            # If we are testing without hardware we just need a few attributes to be set correctly.
-            cls.scanner = None
-            cls.asm = None
-
-            # Use Mocks of the classes to be able to call the fake VAs as for instance mppc.dataContent.value
-            cls.mppc = Mock()
-            cls.mppc.dataContent.value = 'empty'
-
-            cls.multibeam = Mock()
-            cls.multibeam.pixelSize.value = (4.0e-9, 4.0e-9)
-            cls.multibeam.resolution.value = (800, 800)
-
-            cls.descanner = None
-            cls.stage = None
-            cls.ccd = None
-            cls.beamshift = None
-            cls.lens = None
-        else:
-            # get the hardware components from the simulators or hardware
+            try:
+                test.start_backend(FASTEM_CONFIG_ASM)
+            except Exception:
+                logging.info("Failed to start testing, using mocked components for testing.")
+                cls.mocked_testing = True  # If we cannot start the backend, run the mocked test cases.
+                # If we are testing without hardware we just need a few attributes to be set correctly.
+                cls.scanner = None
+                cls.asm = None
```

Comment:
It's quite odd to run a test case differently if the init fails. I'd suggest to have another test case which runs with the mock components. It can inherit from the TestFastEMAcquisitionTask, and override a few of the methods/tests that are not supported. Here, if starting the backend fails, it should just fail!